lib/ansible/module_utils/basic.py

```diff
@@ -2019,8 +2020,8 @@ def _return_formatted(self, kwargs):
             else:
                 self.warn(kwargs['warnings'])
 
-        if warnings.global_warnings:
-            kwargs['warnings'] = warnings.global_warnings
+        if get_warning_messages():
+            kwargs['warnings'] = get_warning_messages()
```

Comment:
Whenever I call a function twice like this (first to check if the value meets a condition and then to make use of that value) I like to turn it into calling the function once and storing the value.  That way the cost of calling the function and any logic the function has to perform are only done once.

``` python

warnings = get_warning_messages()
if warnings:
    kwargs['warnings'] = warnings
```

In this case, we're only saving the cost of calling the function and converting a list into a tuple (and only doing it once per module) so it shouldn't be a big overhead.  But it is nice to do as not everyone will know that the implementation of `get_warning_messages()` is so lightweight (and who knows, it might become a more costly operation in the future)

Reply:
Thanks for this feedback. I had this same question when I was writing it. The cost of calling isn't too high, so is it worth calling once and saving the result? I was hoping you had thoughts. 🧙🏽‍♂️

> ...not everyone will know that the implementation of get_warning_messages() is so lightweight (and who knows, it might become a more costly operation in the future)

This is a very good point. _I_ know it's lightweight, but future readers might not. And it may change in the future. Thank you for the guidance.